optional_plugins/varianter_serialized/avocado_varianter_serialized/__init__.py

```diff
@@ -0,0 +1,135 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: Red Hat Inc. 2018
+# Authors: Amador Pahim <apahim@redhat.com>
+
+import json
+import sys
+
+from six import iteritems
+
+from avocado.core import exit_codes
+from avocado.core import tree
+from avocado.core import varianter
+from avocado.core.output import LOG_UI
+from avocado.core.plugin_interfaces import CLI
+from avocado.core.plugin_interfaces import Varianter
+
+
+class VarianterSerializedCLI(CLI):
+
+    """
+    Serialized based Varianter options
+    """
+
+    name = 'serialized'
+    description = "Serialized based Varianter options for the 'run' subcommand"
+
+    def configure(self, parser):
+
+        for name in ("run", "variants"):  # intentionally omitting "multiplex"
+            subparser = parser.subcommands.choices.get(name, None)
+            if subparser is None:
+                continue
+            sparser = subparser.add_argument_group('json serialized varianter '
+                                                   'options')
+            sparser.add_argument('--load-variants', default=None,
```

Comment:
I'm thinking about the naming here, the `--load-variants` sounds to specific and implies you load variants and that's it. I think this option should match what it does so I'd start with naming our Avocado varianter dump-file by some name (eg. `AvocadoVariantsJSON`, or `AVariants` or so) and then use it across everything here. So for example if you chose AVariants as implementation, then we should have:

```
--avariants-json-dump (generate, export)
--avariants-json
AVariantsJson varianter plugin
```

so it's similar to `--mux-yaml` and `--pict-parameter-file`. I think on first sight it should say I'm yet another plugin to get variants from.

Reply:
Also the `dump` and `load` arguments should go into the same `-h` section.

Reply:
Yes, makes sense. I will think this through.